actors/builtin/miner/miner_actor.go

```diff
@@ -282,13 +282,23 @@ func (a Actor) ProveCommitSector(rt Runtime, params *ProveCommitSectorParams) *a
 		rt.Abortf(exitcode.ErrNotFound, "no precommitted sector %v", sectorNo)
 	}
 
+	if rt.CurrEpoch() > precommit.PreCommitEpoch+MaxSealDuration[precommit.Info.RegisteredProof] || rt.CurrEpoch() < precommit.PreCommitEpoch+PreCommitChallengeDelay {
+		rt.Abortf(exitcode.ErrIllegalArgument, "Invalid ProveCommitSector epoch (cur=%d, pcepoch=%d)", rt.CurrEpoch(), precommit.PreCommitEpoch)
+	}
```

Comment:
`verifySeal`, called immediately below here, has another epoch check. Let's move them next to each other so the constraints are clear.